chromium-x11-122.0.6261.128 build failed in yocto for raspberry pi5

Hi all,
sorry to bother you all,
I am building raspberry pi5 image with meta-raspberrypi commit 1879cb831f4ea3e532cb5ce9fa0f32be917e8fa3 (HEAD -> master, origin/master, origin/HEAD)
Author: Max Stepanov <mast@imast.ru>
Date:   Thu Mar 21 13:50:55 2024 -0400

in docker container ubuntu 22.04

poky  commit f4885e97a6b631e0b9b4ec9f17443cb92794866e (HEAD -> master, origin/master, origin/HEAD)
Author: Richard Purdie <richard.purdie@linuxfoundation.org>
Date:   Wed May 8 14:52:36 2024 +0100

meta-openembedded commit 5778e32eae201072c5dc37c9db67dc1848ffb9de (HEAD -> master, origin/master, origin/HEAD)
Author: Martin Jansa <martin.jansa@gmail.com>
Date:   Mon May 6 01:59:47 2024 +0200

..
meta-browser  commit ab764118ea0d9fb4a0f01fe69190e47244708018 (HEAD -> master, origin/master, origin/HEAD)
Author: Max Ihlenfeldt <max@igalia.com>
Date:   Wed May 8 11:03:07 2024 +0200



but chromium-x11-122.0.6261.128 build failed!

what's the issue?
yesterday, I build  on docker container ubuntu18.04, the same fatal error output.

Thank you!


build output from the first fatal error: 

| ../../chrome/browser/file_select_helper.h:16:10: fatal error: 'components/enterprise/buildflags/buildflags.h' file not found
|    16 | #include "components/enterprise/buildflags/buildflags.h"


....

| In file included from ../../chrome/browser/devtools/devtools_window.cc:28:
| ../../chrome/browser/file_select_helper.h:16:10: fatal error: 'components/enterprise/buildflags/buildflags.h' file not found
|    16 | #include "components/enterprise/buildflags/buildflags.h"
|       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| 1 error generated.


....


I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/nss3 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/nspr -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/dbus-1.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/lib/dbus-1.0/include -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/at-spi2-atk/2.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/at-spi-2.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/libmount -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/blkid -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/atk-1.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/dbus-1.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/lib/dbus-1.0/include -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/include/glib-2.0 -I/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot/usr/lib/glib-2.0/include -Wno-c++11-narrowing-const-reference -std=c++20 -Wno-trigraphs -gsimple-template-names -fno-exceptions -fno-rtti -fvisibility-inlines-hidden  -O2 -pipe  -feliminate-unused-debug-types   -fmacro-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128=/usr/src/debug/chromium-x11/122.0.6261.128  -fdebug-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128=/usr/src/debug/chromium-x11/122.0.6261.128  -fmacro-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/out/Release=/usr/src/debug/chromium-x11/122.0.6261.128  -fdebug-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/chromium-122.0.6261.128/out/Release=/usr/src/debug/chromium-x11/122.0.6261.128  -fdebug-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot=  -fmacro-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot=  -fdebug-prefix-map=/home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/recipe-sysroot-native= -g1 -fvisibility-inlines-hidden -stdlib=libc++ -c ../../chrome/browser/media/router/providers/cast/mirroring_activity.cc -o obj/chrome/browser/media/router/router/mirroring_activity.o
| ninja: build stopped: subcommand failed.
| WARNING: /home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/temp/run.do_compile.392:144 exit 1 from 'ninja -v -j 15 chrome chrome_sandbox chromedriver.unstripped'
| WARNING: Backtrace (BB generated script):
| 	#1: do_compile, /home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/temp/run.do_compile.392, line 144
| 	#2: main, /home/yocto/yocto/source/build/tmp/work/cortexa76-poky-linux/chromium-x11/122.0.6261.128/temp/run.do_compile.392, line 148
NOTE: recipe chromium-x11-122.0.6261.128-r0: task do_compile: Failed
ERROR: Task (/home/yocto/yocto/source/layers/meta-browser/meta-chromium/recipes-browser/chromium/chromium-x11_122.0.6261.128.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 9712 tasks of which 9705 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/yocto/yocto/source/layers/meta-browser/meta-chromium/recipes-browser/chromium/chromium-x11_122.0.6261.128.bb:do_compile
Summary: There was 1 ERROR message, returning a non-zero exit code.
Build step 'Execute shell' marked build as failure

--- a/meta-chromium/recipes-browser/chromium/chromium-gn.inc
+++ b/meta-chromium/recipes-browser/chromium/chromium-gn.inc
@@ -13,6 +13,7 @@ B = "${S}/${OUTPUT_DIR}"
 # Backported patches.
 SRC_URI += "\
     file://backport/Make-toolchain_supports_rust_thin_lto-configurable.patch \
+    file://backport/Add-missing-components-enterprise-buildflag.patch \
 "
 # Non-specific patches.
 SRC_URI += "\

--- a/meta-chromium/recipes-browser/chromium/files/backport/Add-missing-components-enterprise-buildflag.patch
+++ b/meta-chromium/recipes-browser/chromium/files/backport/Add-missing-components-enterprise-buildflag.patch
@@ -1,0 +1,54 @@
+From d695f5ed165f88b157dcc9d150a05ce756405fa7 Mon Sep 17 00:00:00 2001
+From: Takuto Ikuta <tikuta@chromium.org>
+Date: Thu, 25 Apr 2024 07:25:32 +0000
+Subject: [PATCH] Backport "Add missing components/enterprise/buildflags
+ dependency"
+
+This is a backport of two CLs that add missing dependencies to fix a
+build race condition that would sometimes lead to build errors. Both CLs
+will be included in upstream's 126 release.
+
+Upstream-Status: Backport [https://crrev.com/c/5487538, https://crrev.com/c/5526618]
+Signed-off-by: Max Ihlenfeldt <max@igalia.com>
+---
+ chrome/browser/devtools/BUILD.gn      | 1 +
+ chrome/browser/extensions/BUILD.gn    | 1 +
+ chrome/browser/safe_browsing/BUILD.gn | 1 +
+ 3 files changed, 3 insertions(+)
+
+diff --git a/chrome/browser/devtools/BUILD.gn b/chrome/browser/devtools/BUILD.gn
+index 159ee09..fee4c31 100644
+--- a/chrome/browser/devtools/BUILD.gn
++++ b/chrome/browser/devtools/BUILD.gn
+@@ -113,6 +113,7 @@ static_library("devtools") {
+     "//chrome/browser/autofill:autofill",
+     "//components/autofill/content/browser:browser",
+     "//components/autofill/core/browser:browser",
++    "//components/enterprise/buildflags",
+     "//components/paint_preview/buildflags:buildflags",
+     "//content/public/browser",
+     "//net",
+diff --git a/chrome/browser/extensions/BUILD.gn b/chrome/browser/extensions/BUILD.gn
+index 0ca4995..554cc16 100644
+--- a/chrome/browser/extensions/BUILD.gn
++++ b/chrome/browser/extensions/BUILD.gn
+@@ -853,6 +853,7 @@ static_library("extensions") {
+     "//components/embedder_support",
+     "//components/embedder_support:browser_util",
+     "//components/enterprise",
++    "//components/enterprise/buildflags",
+     "//components/favicon/content",
+     "//components/feedback",
+     "//components/gcm_driver",
+diff --git a/chrome/browser/safe_browsing/BUILD.gn b/chrome/browser/safe_browsing/BUILD.gn
+index 4a67fbf..031e76b 100644
+--- a/chrome/browser/safe_browsing/BUILD.gn
++++ b/chrome/browser/safe_browsing/BUILD.gn
+@@ -31,6 +31,7 @@ static_library("safe_browsing") {
+     "//chrome/common:constants",
+     "//components/browser_sync",
+     "//components/enterprise:enterprise",
++    "//components/enterprise/buildflags",
+     "//components/enterprise/common:strings",
+     "//components/keyed_service/content",
+     "//components/language/core/browser",

